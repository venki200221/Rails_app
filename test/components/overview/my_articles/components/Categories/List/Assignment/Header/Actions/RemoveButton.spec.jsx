import React from 'react';
import { shallow } from 'enzyme';
import '../../../../../../../../../testHelper';

import RemoveButton from '../../../../../../../../../../app/assets/javascripts/components/overview/my_articles/components/Categories/List/Assignment/Header/Actions/RemoveButton';

describe('RemoveButton', () => {
  const unassign = jest.fn();
  const props = {
    assignment: {},
    unassign
  };
  const component = shallow(<RemoveButton {...props} />);

  it('renders the button', () => {
    expect(component).toMatchSnapshot();
  });

  it('calls the unassign function when clicked', () => {
    const button = component.find('button');
    button.props().onClick();

    expect(unassign.mock.calls.length).toEqual(1);
  });
});
