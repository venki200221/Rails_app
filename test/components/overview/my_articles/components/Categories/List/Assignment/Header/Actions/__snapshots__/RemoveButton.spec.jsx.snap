// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveButton renders the button 1`] = `ShallowWrapper {}`;
